.github/workflows/php-quality-checks.yml

```diff
@@ -0,0 +1,607 @@
+            ${{ inputs.composer-args }} \
+            ${{ inputs.debug && '--verbose' || '' }}
+
+      # Archive vendor directory for other jobs
+      - name: Archive vendor directory
+        run: tar -czf vendor.tar.gz vendor/
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: vendor-php${{ inputs.php-version }}
+          path: vendor.tar.gz
+          retention-days: 1
+
+  # Composer security audit
+  security-audit:
+    name: 🔒 Security Audit
+    needs: [prepare, install]
+    if: needs.prepare.outputs.has-composer == 'true'
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.prepare.outputs.php-container }}
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Install Composer
+        run: |
+          apk add --no-cache git unzip
+          curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: vendor-php${{ inputs.php-version }}
+
+      - name: Extract vendor directory
+        run: tar -xzf vendor.tar.gz
+
+      - name: Run Composer security audit
+        run: |
+          echo "🔒 Running Composer security audit..."
+          composer audit --format=table ${{ inputs.debug && '--verbose' || '' }}
+
+  # PHPStan static analysis
+  phpstan:
+    name: 🔍 PHPStan Analysis
+    needs: [prepare, install]
+    if: |
+      inputs.skip-phpstan == false && 
+      needs.prepare.outputs.has-composer == 'true'
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.prepare.outputs.php-container }}
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Install system dependencies
+        run: apk add --no-cache git
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: vendor-php${{ inputs.php-version }}
+
+      - name: Extract vendor directory
+        run: tar -xzf vendor.tar.gz
+
+      - name: Cache PHPStan results
+        uses: actions/cache@v4
+        with:
+          path: /tmp/phpstan
+          key: phpstan-php${{ inputs.php-version }}-${{ inputs.phpstan-level }}-${{ hashFiles('**/*.php') }}
+          restore-keys: |
+            phpstan-php${{ inputs.php-version }}-${{ inputs.phpstan-level }}-
+            phpstan-php${{ inputs.php-version }}-
+
+      - name: Run PHPStan analysis
+        run: |
+          echo "🔍 Running PHPStan analysis (Level ${{ inputs.phpstan-level }})..."
+          
+          # Check if PHPStan is installed
+          if [ -f "vendor/bin/phpstan" ]; then
+            phpstan_cmd="vendor/bin/phpstan"
+          else
+            echo "Installing PHPStan globally..."
+            composer global require phpstan/phpstan:^1.0
+            phpstan_cmd="phpstan"
+          fi
+
+          # Build PHPStan command
+          cmd="php -d memory_limit=${{ inputs.memory-limit }} $phpstan_cmd analyse"
+          
+          # Use existing config or set defaults
```

Comment:
There should be an option to use a custom script for phpstan. For example we often have a `phpstan.sh` script in the root of the directory